src/provider/java/adoptiumJavaVersions.ts

```diff
@@ -0,0 +1,36 @@
+import { ADOPTIUM_API } from "#common/constants/urls.ts";
+import { moduleLogger } from "#core/logger.ts";
+import { defineProvider } from "#core/provider.ts";
+import { AdoptiumJavaReleases, AdoptiumJavaRuntimeEntries } from "#schema/java/adoptiumJavaData.ts";
+
+const RUNTIMES_URL = new URL("v3/", ADOPTIUM_API);
+
+const logger = moduleLogger();
+
+export default defineProvider({
+	id: "adoptium-java",
+
+	async provide(http): Promise<(AdoptiumJavaRuntimeEntries | null)[]> {
+		const releases = AdoptiumJavaReleases.parse(
+			(await http.getCached(new URL("info/available_releases", RUNTIMES_URL), "available-releases.json")).json()
+		);
+
+		return Promise.all(releases.available_releases.map(async version => {
+			const options = new URLSearchParams({
+				image_type: "jre",
+			});
+
+			const response = await http.getCached(new URL(`assets/feature_releases/${version}/ga?${options}`, RUNTIMES_URL), `adoptium-java-runtime-${version}.json`)
+				.catch(() => {
+					logger.error(`Failed to get Adoptium JRE ${version} with General Access Version`);
+					return null;
+				});
```

Comment:
TODO comment would be nice
```suggestion
			const response = await http.getCached(new URL(`assets/feature_releases/${version}/ga?${options}`, RUNTIMES_URL), `adoptium-java-runtime-${version}.json`)
				.catch(() => {
					// TODO: don't ignore all errors
					logger.error(`Failed to get Adoptium JRE ${version} with General Access Version`);
					return null;
				});
```

Reply:
Would this be better with hardcoded? Because Adoptium doesn't have GA release with Java 16, even it is exist in `info/available_releases` endpoint.

Reply:
I wouldn't want us to need to edit it all the time
There might be another solution but I'm not aware of it

Reply:
We might still need to add more exclusions in future
In our current meta we just ignore a version if the status code is 404